bitswap/client/client.go

```diff
@@ -97,6 +98,27 @@ func WithoutDuplicatedBlockStats() Option {
 	}
 }
 
+// WithContentRouting enables content.Routing lookups to find providers for
+// the content to be retrieved using Bitswap.
+func WithContentRouting(router routing.Routing) Option {
+	return func(bs *Client) {
+		bs.router = router
+	}
+}
```

Comment:
I find this problematic as it will silently disable providerQuery-ing if users do not remember to pass in the new option with a content router.

I would prefer to break New() and require a new router there (supporting that it is nil).

Reply:
> I would prefer to break New() and require a new router there (supporting that it is nil).

That's fine by me. In general I prefer options to passing nil's but for new consumers (as opposed to ones who will migrate during the breaking change to bsnet) I can see the value in making them think about whether they want to provide a content router or not.